src/registry/search.rs

```diff
@@ -17,7 +40,96 @@ pub struct RegistrySearchArgs {
     #[arg(long, default_value = "false")]
     lineage: bool,
 
-    /// The telemetry schema containing the versions (url or file)
-    #[arg(short, long)]
-    pub schema: Option<String>,
+    /// Parameters to specify the diagnostic format.
+    #[command(flatten)]
+    pub diagnostic: DiagnosticArgs,
+}
+
+#[derive(thiserror::Error, Debug, serde::Serialize, Diagnostic)]
+enum Error {
+    #[error("{0}")]
+    StdIoError(String),
+}
+
+impl From<std::io::Error> for Error {
+    fn from(e: std::io::Error) -> Self {
+        Error::StdIoError(e.to_string())
+    }
+}
+
+// Our search application state
+struct SearchApp<'a> {
+    schema: &'a ResolvedTelemetrySchema,
+}
+
+impl<'a> SearchApp<'a> {
+    fn new(schema: &'a ResolvedTelemetrySchema) -> SearchApp<'a> {
+        SearchApp { schema }
+    }
+
+    fn render(&self, frame: &mut Frame<'_>) {
+        let area = frame.size();
+        frame.render_widget(
+            Paragraph::new(format!(
+                "Hello Search {0}! (press 'q' to quit)",
+                self.schema.schema_url
+            ))
+            .white()
+            .on_blue(),
+            area,
+        );
+    }
+    // Returns true when it's time to quit.
+    fn process(&mut self, event: Event) -> Result<bool, Error> {
+        if let Event::Key(key) = event {
+            if key.kind == KeyEventKind::Press && key.code == KeyCode::Char('q') {
+                return Ok(true);
+            }
+        }
+        Ok(false)
+    }
+}
+
+fn run_ui(schema: &ResolvedTelemetrySchema) -> Result<(), Error> {
+    let mut app = SearchApp::new(schema);
+    let _ = stdout().execute(EnterAlternateScreen)?;
+    enable_raw_mode()?;
+    let mut terminal = Terminal::new(CrosstermBackend::new(stdout()))?;
+    terminal.clear()?;
+
+    // main loop
+    loop {
+        let _ = terminal.draw(|frame| app.render(frame))?;
+        if event::poll(std::time::Duration::from_millis(16))? {
+            if app.process(event::read()?)? {
+                break;
+            }
+        }
+    }
+
+    let _ = stdout().execute(LeaveAlternateScreen)?;
+    disable_raw_mode()?;
+    Ok(())
+}
+
+pub(crate) fn command(
+    logger: impl Logger + Sync + Clone,
+    cache: &Cache,
+    args: &RegistrySearchArgs,
+) -> Result<(), DiagnosticMessages> {
+    logger.loading(&format!("Resolving registry `{}`", args.registry.registry));
+
+    let registry_id = "default";
+    let registry_path =
+        semconv_registry_path_from(&args.registry.registry, &args.registry.registry_git_sub_dir);
+
+    // Load the semantic convention registry into a local cache.
+    let semconv_specs = load_semconv_specs(&registry_path, cache, logger.clone())?;
+    let mut registry = SemConvRegistry::from_semconv_specs(registry_id, semconv_specs);
+    let schema = resolve_semconv_specs(&mut registry, logger.clone())?;
+
+    // TODO - We should have two modes:
+    // 1. An interactive UI
```

Comment:
For the interactive mode, see this commit https://github.com/open-telemetry/weaver/commit/96e2c134214c237cf3c70e57b90ed7dbd08763d1 path=src/search.